pkg/provider/consulcatalog/consul_catalog.go

```diff
@@ -48,9 +57,14 @@ type Provider struct {
 	Cache             bool            `description:"Use local agent caching for catalog reads." json:"cache,omitempty" toml:"cache,omitempty" yaml:"cache,omitempty" export:"true"`
 	ExposedByDefault  bool            `description:"Expose containers by default." json:"exposedByDefault,omitempty" toml:"exposedByDefault,omitempty" yaml:"exposedByDefault,omitempty" export:"true"`
 	DefaultRule       string          `description:"Default rule." json:"defaultRule,omitempty" toml:"defaultRule,omitempty" yaml:"defaultRule,omitempty"`
+	ConnectAware      bool            `description:"Enable Consul Connect support." json:"connectAware,omitempty" toml:"connectAware,omitempty" yaml:"connectAware,omitempty"`
+	ConnectByDefault  bool            `description:"Automatically connect to a service via Consul connect." json:"connectByDefault,omitempty" toml:"connectByDefault,omitempty" yaml:"connectByDefault,omitempty"`
+	ServiceName       string          `description:"Name of the traefik service in Consul Catalog." json:"serviceName,omitempty" toml:"serviceName,omitempty" yaml:"serviceName,omitempty"`
```

Comment:
Because more than one traefik could run against one consul (probably with different prefixes)

Reply:
I suspected as much, but I don't have much knowledge about the consul ecosystem itself so I wanted to be sure. fair enough.
Although, in what kind of scenario would it make sense to have several traefik instances here? Is it a load-balancing of several traefik instances doing the same thing? Or is it independent instances doing a completely different job (and why)? (I'm not asking for a justification, I believe you, it's just out of curiosity).

Reply:
For load balancing the traefik instances would have the same service name. Personally I currently just have one traefik; but I could imagine that it makes sense to start a second one to test out new things etc. Also it might make sense (risk-wise) to have multiple traefik instances that can only access a certain set of services (via consul connect intentions). Either way, I think it is little enough code that I'd rather keep it and not prevent some valid scenarios.